### PR TITLE
add dkd

### PR DESCRIPTION
**Pull request by WangChen0902**
backend=pil, alpha=1.0, beta=1.0：top1=0.7259
backend=pil, alpha=1.0, beta=0.5: top1=0.7233 (repo配置）
backend=cv2, alpha=1.0, beta=0.5: top1=0.7231 (repo配置）
原代码精度: 71.75

### CODE REVIEW

File: ppcls/loss/dkdloss.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+
+
+class DKDLoss(nn.Layer):
+    """
+    DKDLoss
```
Comment (littletomatodonkey) [inline ppcls/loss/dkdloss.py:8]: 1. 新建一个dkdloss.py，输入为tensor，并添加论文地址和参考代码的引用
2. 这个实现写distillationloss.py里面，继承自上面文件实现的类，参考下dmlloss和distillationdmlloss

File: ppcls/loss/distillationloss.py [hunk 2]
```diff
@@ -204,3 +205,33 @@ def forward(self, predicts, batch):
...
+        loss_dict = dict()
+        for idx, pair in enumerate(self.model_name_pairs):
+            out1 = predicts[pair[0]]
+            out2 = predicts[pair[1]]
+            if self.key is not None:
+                out1 = out1[self.key]
+                out2 = out2[self.key]
+            loss = super().forward(out1, out2, batch)
+            loss_dict[self.name] = loss
+        return loss_dict
```
Comment (littletomatodonkey) [inline ppcls/loss/distillationloss.py]: 提交之前做一下pre-commit

File: ppcls/loss/dkdloss.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    paper: https://arxiv.org/abs/2203.08679
+    code: https://github.com/megvii-research/mdistiller
```
Comment (littletomatodonkey) [inline ppcls/loss/dkdloss.py]: 改成下面这样

```
Reference: https://arxiv.org/abs/2203.08679
Code was heavily based on https://github.com/megvii-research/mdistiller
```

File: ppcls/loss/distillationloss.py [hunk 4]
```diff
@@ -204,3 +205,33 @@ def forward(self, predicts, batch):
...
+    def forward(self, predicts, batch):
+        loss_dict = dict()
+        for idx, pair in enumerate(self.model_name_pairs):
+            out1 = predicts[pair[0]]
+            out2 = predicts[pair[1]]
+            if self.key is not None:
+                out1 = out1[self.key]
+                out2 = out2[self.key]
+            loss = super().forward(out1, out2, batch)
+            loss_dict[self.name] = loss
```
Comment (littletomatodonkey) [inline ppcls/loss/distillationloss.py]: `loss_dict[self.name]` -> `loss_dict[f"{self.name}_{pair[0]}_{pair[1]}]`

避免命名重复

**Review by littletomatodonkey — APPROVED** [review]
